lib/shopify_api_console/console.rb

```diff
@@ -179,5 +185,11 @@ def default?(connection)
     def no_config_file_error(filename)
       raise ConfigFileError, "There is no config file at #{filename}"
     end
+
+    def shopify_api_latest_version
+      quarter = 3
+      latest_released_month = SHOPIFY_API_RELEASE_MONTHS[(Time.now.month - 1)/ quarter]
```

Comment:
This is a neat trick!

But that said, while stable API versions typically do get released on the first of the month/quarter, sometimes (as with version 2020-07) the version was released a day later).

Perhaps a safer alternative to computing the version handle in this library would be:

Since you have the shop domain + API credentials from above, perhaps you should call the GraphQL Admin API to obtain the latest API version from https://shopify.dev/api/admin/graphql/reference/common-objects/queryroot#publicapiversions-2021-07.

The last `supported: true` version would be the one to pick here.

![image](https://user-images.githubusercontent.com/1916444/129572268-a09a5dc3-86c6-45ab-a582-11f226f7ce61.png)

Reply:
Hi @chrisbutcher,

Thanks for review the changes. It is good to know that we can query the supported API version. Your suggestion is better than the time-induction approach. It helps to avoid timing sensitivity (especially in different timezones).

I will implement the GraphQL query and update this PR soon. I am quite tired with providing the API version every time I open the console.